test/facility-investor/post-facility-investor.api-test.ts

```diff
@@ -0,0 +1,261 @@
+import { PROPERTIES } from '@ukef/constants';
+import { withAcbsAuthenticationApiTests } from '@ukef-test/common-tests/acbs-authentication-api-tests';
+import { IncorrectAuthArg, withClientAuthenticationTests } from '@ukef-test/common-tests/client-authentication-api-tests';
+import { withRequiredDateOnlyFieldValidationApiTests } from '@ukef-test/common-tests/request-field-validation-api-tests/required-date-only-field-validation-api-tests';
+import { withRequiredNonNegativeNumberFieldValidationApiTests } from '@ukef-test/common-tests/request-field-validation-api-tests/required-non-negative-number-field-validation-api-tests';
+import { withStringFieldValidationApiTests } from '@ukef-test/common-tests/request-field-validation-api-tests/string-field-validation-api-tests';
+import { Api } from '@ukef-test/support/api';
+import { ENVIRONMENT_VARIABLES } from '@ukef-test/support/environment-variables';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import nock from 'nock';
+
+describe('POST /facilities/{facilityIdentifier}/investors', () => {
+  const valueGenerator = new RandomValueGenerator();
+  const portfolioIdentifier = PROPERTIES.GLOBAL.portfolioIdentifier;
+  const facilityIdentifier = valueGenerator.stringOfNumericCharacters({ length: 10 });
+  const createFacilityInvestorUrl = `/api/v1/facilities/${facilityIdentifier}/investors`;
+
+  const sectionIdentifier = PROPERTIES.FACILITY_INVESTOR.DEFAULT.sectionIdentifier;
+  const facilityStatusCode = PROPERTIES.FACILITY_INVESTOR.DEFAULT.facilityStatus.facilityStatusCode;
+  const involvedPartyIdentifier = PROPERTIES.FACILITY_INVESTOR.DEFAULT.involvedParty.partyIdentifier;
+  const effectiveDateInFuture = '9999-01-02';
+  const guaranteeExpiryDateInFuture = '9999-12-31';
+  const lenderType = valueGenerator.stringOfNumericCharacters({ length: 3 });
+  const currency = 'GBP';
+  const maximumLiability = 12345.6;
+  const limitTypeCode = valueGenerator.stringOfNumericCharacters({ minLength: 1, maxLength: 2 });
+  const limitKey = valueGenerator.stringOfNumericCharacters({ maxLength: 10 });
+
+  const requestBodyToCreateFacilityInvestor = [
```

Comment:
Can we please have `requestBodyInvalidToCreateFacilityInvestor` for few negative assertions please?

Reply:
The `with<Description>FieldValidationApiTests` function calls lower down inject tests that check invalid request bodies for the negative assertions. The full test suite result is
```
POST /facilities/{facilityIdentifier}/investors
    √ returns a 500 response if creating a session with the IdP fails (298 ms)
    √ returns a 500 response if getting an id token from the IdP fails (75 ms)
    √ returns a 500 response if getting an id token from the IdP times out (47 ms)
    √ returns a 500 response if creating a session with the IdP times out (13 ms)
    √ returns a 401 response when the API key header is missing (49 ms)
    √ returns a 401 response when the API key header name is incorrect (12 ms)
    √ returns a 401 response when the API key header value is incorrect (10 ms)
    √ returns a 401 response when the API key header value is empty (11 ms)
    √ creates a facility party in ACBS and returns a 201 response with the facility identifier if the creation in ACBS is successful (94 ms)
    √ creates a facility party in ACBS with a default lenderType if it is not specified in the request (85 ms)
    √ returns a 404 response if ACBS responds with a 400 response that is a string containing "The facility not found" (83 ms)
    √ returns a 400 response if ACBS responds with a 400 response that is not a string (81 ms)
    √ returns a 400 response if ACBS responds with a 400 response that is a string that does not contain "The facility not found" (80 ms)
    √ returns a 500 response if ACBS responds with an error code that is not 400" (90 ms)
    effectiveDate validation
      √ returns a 400 response if effectiveDate is not present (15 ms)
      √ returns a 400 response if effectiveDate has time part of date string (10 ms)
      √ returns a 400 response if effectiveDate is not in YYYY-MM-DD date format (7 ms)
      √ returns a 400 response if effectiveDate is not a valid date (9 ms)
      √ returns a 400 response if effectiveDate is not a real day (7 ms)
      √ returns a 201 response if effectiveDate is a valid date (79 ms)
    guaranteeExpiryDate validation
      √ returns a 400 response if guaranteeExpiryDate is not present (9 ms)
      √ returns a 400 response if guaranteeExpiryDate has time part of date string (9 ms)
      √ returns a 400 response if guaranteeExpiryDate is not in YYYY-MM-DD date format (12 ms)
      √ returns a 400 response if guaranteeExpiryDate is not a valid date (6 ms)
      √ returns a 400 response if guaranteeExpiryDate is not a real day (10 ms)
      √ returns a 201 response if guaranteeExpiryDate is a valid date (77 ms)
    maximumLiability validation
      √ returns a 400 response if maximumLiability is not present (6 ms)
      √ returns a 400 response if maximumLiability is less than 0 (5 ms)
      √ returns a 201 response if maximumLiability is 0 (68 ms)
      √ returns a 201 response if maximumLiability is greater than 0 (99 ms)
    currency validation
      √ returns a 400 response if currency is not present (5 ms)
      √ returns a 400 response if currency is an empty string (5 ms)
      √ returns a 400 response if currency has fewer than 3 characters (5 ms)
      √ returns a 400 response if currency has more than 3 characters (4 ms)
      √ returns a 201 response if currency has 3 characters (102 ms)
    facilityIdentifier validation
      √ returns a 400 response if facilityIdentifier is not present (10 ms)
      √ returns a 400 response if facilityIdentifier is an empty string (13 ms)
      √ returns a 400 response if facilityIdentifier has fewer than 10 characters (11 ms)
      √ returns a 400 response if facilityIdentifier has more than 10 characters (12 ms)
      √ returns a 201 response if facilityIdentifier has 10 characters (92 ms)
    lenderType validation
      √ returns a 201 response if lenderType is not present (84 ms)
      √ returns a 400 response if lenderType is an empty string (8 ms)
      √ returns a 400 response if lenderType has fewer than 3 characters (7 ms)
      √ returns a 400 response if lenderType has more than 3 characters (6 ms)
      √ returns a 201 response if lenderType has 3 characters (81 ms)
```